source/Applications/Advanced/HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cs

```diff
@@ -0,0 +1,157 @@
+
+            Console.WriteLine("Reading end-effector pose in robot base frame");
+            var endEffectorPoseFile = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData)
+                               + "/Zivid/RobotTransform.yaml";
+
+            var eyeInHandTransformation = readTransform(eyeInHandTransformationFile);
+            var endEffectorPose = readTransform(endEffectorPoseFile);
+
+            // Converting to Mat.NET matrices for easier computation
+            Matrix<float> eyeInHandMatrix = zividToMathDotNet(eyeInHandTransformation);
+            Matrix<float> endEffectorMatrix = zividToMathDotNet(endEffectorPose);
+
+            var dataFile = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData) + "/Zivid/ZividGem.zdf";
+            Console.WriteLine("Reading ZDF frame from file: " + dataFile);
+            var frame = new Zivid.NET.Frame(dataFile);
+            var pointCloud = frame.PointCloud;
+
+            Console.WriteLine("Computing camera pose in robot base frame");
+            Matrix<float> transformBaseToCamera = endEffectorMatrix * eyeInHandMatrix;
+
+
+            switch (Interaction.EnterCommand())
+            {
+                case CommandType.TransformSinglePoint:
+
+                    Console.WriteLine("Transforming single point");
+
+                    // The (picking) point is defined as image coordinates in camera frame. It is hard-coded for the
+                    // ZividGem.zdf
+                    const int imageCoordinateX = 1357;
+                    const int imageCoordinateY = 666;
+                    var xyz = pointCloud.CopyPointsXYZW();
+
+                    var pickingPoint = new float[]{xyz[imageCoordinateY, imageCoordinateX, 0],
+                                                            xyz[imageCoordinateY, imageCoordinateX, 1],
+                                                            xyz[imageCoordinateY, imageCoordinateX, 2],
+                                                            xyz[imageCoordinateY, imageCoordinateX, 3]};
+
+                    Vector<float> pointInCameraFrame = CreateVector.DenseOfArray(pickingPoint);
+                    Console.WriteLine("Point coordinates in camera frame: " + pointInCameraFrame);
+
+                    Console.WriteLine("Transforming (picking) point from camera to robot base frame");
+                    Vector<float> pointInBaseFrame = transformBaseToCamera * pointInCameraFrame;
+
+                    Console.WriteLine("Point coordinates in robot base frame: " + pointInBaseFrame);
+                    break;
+
+                case CommandType.TransformPointCloud:
+
+                    Console.WriteLine("Transforming point cloud");
+                    var transformMatrix = mathDotNetToZivid(transformBaseToCamera);
+                    pointCloud.Transform(transformMatrix);
+
+                    var saveFile = "ZividGemTransformed.zdf";
```

Comment:
The var name is not consistent with what we use in other samples. I think we use fileName don't we?

Reply:
We use dataFile, but I already use that var name when reading point cloud, hence I need a new name for the saved file. I can change it til fileName instead of saveFile

Reply:
I don't like 'saveFile' - it sounds like imperative. How about dataFile and dataFileTransformed / dataTransformedFile / transformedDataFile?

Reply:
That is a better solution! I like the name "dataFileTransformed" as it resonates with what we call the file, ZividGem -> ZividGemTransformed and dataFile -> dataFileTransformed

Reply:
Agree, still to be implemented.

Reply:
done